src/nix-env/nix-env.cc

```diff
@@ -527,6 +535,11 @@ static void installDerivations(Globals & globals,
         newNames.insert(DrvName(i.queryName()).name);
     }
 
+    if (priority) {
+        for (auto & drv : newElems) {
+            setMetaFlag(*globals.state, drv, "priority", std::to_string((priority.value())));
```

Comment:
Still getting stored as a string - that has parity with `opSetFlag`. If you want to store as an int here, i can do that, but id also want to update `opSetFlag` to match the behavior (i.e. have a special case for the priority flag to parse it and store as an int).